### PR TITLE
(MAINT) Add construct-root-routes-handler fn

### PR DESCRIPTION
**Pull request by fpringvaldsen**
Extract the creation of the ruby request handler and the
clojure request wrapper, as well as their use in the creation of
the root-routes handler, into a separate function,
construct-root-routes-handler.

### CODE REVIEW

**Comment by haus** [comment]
This looks pretty reasonable to me.

File: src/clj/puppetlabs/services/master/master_core.clj [hunk 1]
```diff
@@ -549,4 +550,28 @@
    (let [handler-maybe-with-authorization (if use-legacy-auth-conf
                                             route-handler
                                             (authorization-fn route-handler))]
-     (wrap-middleware handler-maybe-with-authorization puppet-version))))
\ No newline at end of file
+     (wrap-middleware handler-maybe-with-authorization puppet-version))))
+
+(schema/defn ^:always-validate
+  construct-root-routes-handler :- bidi-schema/RoutePair
```
Comment (haus) [inline src/clj/puppetlabs/services/master/master_core.clj]: nitpick: this doesn't really produce a handler. the `comidi/routes->handler` call does that later on.

Comment (fpringvaldsen) [inline src/clj/puppetlabs/services/master/master_core.clj]: Ah, okay. Do you know what it does produce?

Comment (haus) [inline src/clj/puppetlabs/services/master/master_core.clj]: just a route tree. i think calling it `construct-root-routes` makes sense to me.

**Comment by fpringvaldsen** [comment]
@haus I've changed the name of that function

**Comment by haus** [comment]
:+1:

**Comment by KevinCorcoran** [comment]
What is the reason for this change?  (I assume it's so that this code will work in PE.)  If that's so, It would probably be worthwhile to update the commit message to explain that.  I'm not usually a stickler for that kind of thing, but in this case, it's not at all obvious at a glance why this change is necessary.

**Comment by KevinCorcoran** [comment]
:+1: looks good except for perhaps updating commit message / PR description

**Comment by KevinCorcoran** [comment]
voomed into `2.2.2-master-20160224_010132-g5a62912`

**Pull request merged**